jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessorTest.java

```diff
@@ -704,13 +721,34 @@ public void testGetBaseImage_projectHigherThanJava11() {
   }
 
   @Test
-  public void testGetBaseImage_incompatibleJava8BaseImage() {
+  public void testGetJavaContainerBuilderWithBaseImage_nonRegistry()
+      throws IncompatibleBaseImageJavaVersionException, IOException, InvalidImageReferenceException,
+          CacheDirectoryCreationException {
+    Mockito.when(rawConfiguration.getFromImage()).thenReturn(Optional.of("tar:///path/to.tar"));
+    ImageConfiguration result = getCommonImageConfiguration();
+    Assert.assertEquals(Paths.get("/path/to.tar"), result.getTarPath().get());
+  }
+
+  @Test
+  public void testGetJavaContainerBuilderWithBaseImage_registry()
+      throws IncompatibleBaseImageJavaVersionException, InvalidImageReferenceException, IOException,
+          CacheDirectoryCreationException {
+    Mockito.when(rawConfiguration.getFromImage()).thenReturn(Optional.of("ima.ge/name"));
+    ImageConfiguration result = getCommonImageConfiguration();
+    Assert.assertEquals("ima.ge/name", result.getImage().toString());
+    Assert.assertFalse(result.getDockerClient().isPresent());
```

Comment:
I think we also need `Assert.assertFalse(result.getTarPath().isPresent())`, since a tar image configuration returns a non-null `ImageReference`. That said, I became curious as to what exactly `getImage()` will return for a tar base image?

Reply:
Right now it's just `scratch` as a placeholder; there's a TODO for this when we figure out what to do about the whole exclusive parameter thing in `ImageConfiguration`.